### ISSUE TITLE
sulu_preview.public_preview does not exist

### ISSUE THREAD

**Issue by exastion** [sulu/sulu#6479]
| Q | A
| --- | ---
| Sulu Version | 2.x
| PHP Version | 8.1.2
| DB Version | MariaDB 10.5.11
| Browser Version | Firefox 96.0.2 (64-bit)

# Actual Behavior
I am able to type in the username and password on the admin panel but then the circle does not stop turning.
In the server log I get the following:
`127.0.0.1:40718 [500]: GET /admin/config`
The Debug Toolbar does not show anything in the AJAX requests, but by using the history I get to the Profile in question with the following exception:
`Unable to generate a URL for the named route "sulu_preview.public_preview" as such route does not exist.`

```
Router->generate('sulu_preview.public_preview', array('token' => ':token'), 0) in src/Sulu/Bundle/PreviewBundle/Admin/PreviewAdmin.php (line 66)

                    'start' => $this->urlGenerator->generate('sulu_preview.start'),
                    'render' => $this->urlGenerator->generate('sulu_preview.render'),
                    'update' => $this->urlGenerator->generate('sulu_preview.update'),
                    'update-context' => $this->urlGenerator->generate('sulu_preview.update-context'),
                    'stop' => $this->urlGenerator->generate('sulu_preview.stop'),
                    'preview-link' => $this->urlGenerator->generate('sulu_preview.public_preview', ['token' => ':token'], RouterInterface::ABSOLUTE_URL),
                ],
                'debounceDelay' => $this->previewDelay,
                'mode' => $this->previewMode,
                'audienceTargeting' => \array_key_exists('SuluAudienceTargetingBundle', $this->bundles),
            ];

```

# Expected Behavior

It should work.

# Steps to Reproduce

`npm install`
`npm run buid`
`bin/console sulu:build dev`

Log in to the admin interface

# Possible Solutions

As a workaround, I removed the offending line in `PreviewAdmin.php` to test the thing I actually wanted to test.

The route in question exists in the file `src/Sulu/Bundle/PreviewBundle/Resources/config/routing_public.yml`. I do not know if this file is not included correctly or if something else is the problem.

**Comment by alexander-schranz** [comment]
Can it by that your skeleton was not updated when you did upgrade from Sulu 2.3 to 2.4. There need some new routes be registered then:

https://github.com/sulu/sulu/blob/2.4/UPGRADE.md#added-previewlink-resource-to-previewbundle

Can you try if that solve your problem?

**Comment by exastion** [comment]
Oh, I was not aware that you need to follow the UPGRADE.md also for the core repo itself. Does this mean that if I clone the repo and checkout 2.x I have to do all the Version 2 upgrades?

**Comment by alexander-schranz** [comment]
Oh I see - no that should not be the case. But seems like we did forget to adjust the files here in sulu/sulu do you want to create a pull request adjusting the files in sulu core security.yaml and sulu_admin.yaml the same way how it was done in the skeleton here: https://github.com/sulu/skeleton/pull/139/files

**Comment by exastion** [comment]
I will do that.

**Comment by niklasnatter** [comment]
Fixed in #6483. Thanks for reporting and working on this!
